### PR TITLE
feat: introduce segment/regional prompt

### PR DESCRIPTION
**Pull request by rvaibhav**
- Added new feature of regionwise text prompts alongside global text prompt
- Regional text prompts must be paired with a bounding box description or a segmentation mask of the area of the video that the regional text prompt applies to

### CODE REVIEW

**Comment by rvaibhav** [comment]
**Command Used to generate**
`PYTHONPATH=$(pwd) CHECKPOINT_DIR="/config/models/cosmos-transfer1" python cosmos_transfer1/diffusion/inference/transfer.py \
    --checkpoint_dir /config/models/cosmos-transfer1 \
    --video_save_folder outputs/example1_single_control_depth \
    --controlnet_specs assets/inference_cosmos_transfer1_single_control_regional_prompt.json \
    --offload_text_encoder_model --sigma_max 80 --offload_guardrail_models`
    
 **Output Generated**
 

https://github.com/user-attachments/assets/c7562306-fc2d-4b82-a258-d99baf652a27

File: cosmos_transfer1/diffusion/module/attention.py [hunk 1]
```diff
@@ -137,6 +137,151 @@ class BaseAttentionOp(nn.Module):
...
+        for i in range(len(regional_k)):
+            mask = region_masks[i]
+            mask = mask.to(q.device)
+            if mask.shape[0] != seq_len:
+                mask = preprocess_mask(mask)
+                mask = rearrange(mask, "b t h w ->  b (t h w)")
+            processed_masks.append(mask)
+            encoder_seq_lens.append(regional_k[i].shape[0])
+
+        total_encoder_seq_len = sum(encoder_seq_lens)
```
Comment (tiffanycai6) [inline cosmos_transfer1/diffusion/module/attention.py:243]: Please see the latest update !3100 for changes to logic in this function

Comment (rvaibhav) [reply cosmos_transfer1/diffusion/module/attention.py]: This has been addressed. @tiffanycai6 Can you please take a look again.

Thanks!

**Review by tiffanycai6 — COMMENTED** [review]
Thank you for porting this, great work Vaibhav!

File: cosmos_transfer1/diffusion/inference/transfer.py [hunk 2]
```diff
@@ -191,6 +190,17 @@ def demo(cfg, control_inputs):
     """
 
     control_inputs = validate_controlnet_specs(cfg, control_inputs)
+    region_definitions = None
+    if args.regional_prompts:
+        log.info(f"regional_prompts: {args.regional_prompts}")
+        if args.regional_prompts[0]["region_definitions_path"]:
+            log.info(f"region_definitions_path: {args.regional_prompts[0]['region_definitions_path']}")
+            region_definition_path = args.regional_prompts[0]['region_definitions_path']
```
[the remark this answers is not in the document]
Comment (rvaibhav) [reply cosmos_transfer1/diffusion/inference/transfer.py]: This has been fixed in latest version of code

File: cosmos_transfer1/diffusion/inference/transfer.py [hunk 3]
```diff
@@ -229,6 +239,8 @@ def demo(cfg, control_inputs):
         upsample_prompt=cfg.upsample_prompt,
         offload_prompt_upsampler=cfg.offload_prompt_upsampler,
         process_group=process_group,
+        regional_prompts=[cfg.regional_prompts[0]["prompt"]],
```
[the remark this answers is not in the document]
Comment (rvaibhav) [reply cosmos_transfer1/diffusion/inference/transfer.py]: This is also fixed in latest version of code

File: examples/inference_cosmos_transfer1_7b_regional_prompt.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# Transfer Inference Example: Single Control (Depth)
```
[the remark this answers is not in the document]
Comment (rvaibhav) [reply examples/inference_cosmos_transfer1_7b_regional_prompt.md:1]: ack

**Review by pjannaty — APPROVED** [review]
LGTM. Thank you @rvaibhav and @tiffanycai6!
